handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -614,21 +614,32 @@ public final SSLEngineResult wrap(
                 }
             }
 
-            if (dst.remaining() < MAX_ENCRYPTED_PACKET_LENGTH) {
+            int endOffset = offset + length;
+            long maxEncryptedLen = MAX_ENCRYPTION_OVERHEAD_LENGTH;
+
+            for (int i = offset; i < endOffset; ++i) {
+                final ByteBuffer src = srcs[i];
+                if (src == null) {
+                    throw new IllegalArgumentException("srcs[" + i + "] is null");
```

Comment:
@trustin its an `IllegalArgumentException` as stated in the SSLEngine javadocs: http://docs.oracle.com/javase/7/docs/api/javax/net/ssl/SSLEngine.html#wrap(java.nio.ByteBuffer[],%20java.nio.ByteBuffer)